xlators/mgmt/glusterd/src/glusterd-svc-helper.c

```diff
@@ -761,7 +761,7 @@ __glusterd_send_svc_configure_req(glusterd_svc_t *svc, int flags,
     }
 
     conn = &rpc->conn;
-    if (!conn->connected || conn->disconnected) {
+    if (is_rpc_clnt_disconnected(conn)) {
```

Comment:
nit: I wonder why sometimes conn->disconnected is accessed only under lock (such as the case in is_rpc_clnt_disconnected() above) and sometimes, just as is (send_attach_req() for example).